databind.json/src/databind/json/converters.py

```diff
@@ -167,7 +166,6 @@ def _length_check() -> None:
             try:
                 return python_type(values)
             except TypeError:
-                assert not isinstance(values, types.GeneratorType), (type(values), python_type)
```

Comment:
This gave an Unreachable error in mypy, probably it's due to the newest mypy version knowing to check this